### ISSUE TITLE
ImportError: cannot import name 'abs'

### ISSUE THREAD

**Issue by jslurm** [deepfakes/faceswap-playground#209]
Up and running on a GTX 1070, with Windows 10. I was basically experimenting with different models. Tried GAN (64) and training seemed to be going alright. Decided to try training with perceptual loss, and got the following error: `ModuleNotFoundError: No module named 'keras_vggface'`

So I tried `pip install keras_vggface`, but it looks like that also installed Tensorflow 1.10.0. Now when I try to run any faceswap script, I get `ImportError: cannot import name 'abs'`. I tried installing Tensorflow 1.6.0 again, then ran the install of the entire requirements.txt file, but nothing seems to work.

I'm not very knowledgeable on this stuff & mostly just trying it out, so I'm not really sure what I mucked up or how to fix it. Wondering if anyone might know how I can get back on track (and hopefully try GAN training with perceptual loss)

**Comment by torzdf** [comment]
1.10 of Tensorflow should work with latest FaceSwap commit. I run 1.9 at the moment and it runs fine.

You have 2 options

1) Blow away your virtual environment and recreate
2) from https://github.com/tensorflow/tensorflow/issues/20778

The full solution to this issue is to clean up all TF installation files from your system. Here is how to do that. Run the following first:

`python3 -c 'import tensorflow as tf; print(tf.__file__)'`

To get:
`'path/to/your/python/installation/site-packages/tensorflow/__init__.pyc'`

From the above command, you need to go and remove the folder shown.

Finally, you need to run

```
$ pip uninstall tensorflow # Also add -gpu if you installed the GPU version
$ pip install tensorflow # Also add -gpu if you installed the GPU version
```

Then it should work.

**Comment by jslurm** [comment]
I tried running `python3 -c 'import tensorflow as tf; print(tf.__file__)'` I get an error `'python3' is not recognized as an internal or external command,
operable program or batch file.`

If I try the command with just "python", I get this error:
`File "<string>", line 1`
>`    'import`
>>`          ^`

`SyntaxError: EOL while scanning string literal`

I'm running in an Anaconda prompt, so not sure if that makes a difference in commands to use.

**Comment by jslurm** [comment]
I tried the -pl switch again, and this time it looks like (while initializing) it automatically downloaded keras_vggface (or a trainer? I'm not sure). I had to reduce the batch size, but it looks like that training now works for me. No idea what changed.

I do have a couple more GAN related questions. Other discussions mention that GAN isn't quite ready (or may be broken). That was a few months ago so I'm just curious if there's been any progress on that front? Also, [another discussion mentioned](https://github.com/deepfakes/faceswap-playground/issues/118) training without perceptual loss for 10k iterations, then training with perceptual loss for another 10k iterations, then to "turn on mask refinement." I don't see an option for mask refinement anywhere in the training. Is there a way to do this currently, or is it something that's been removed?
